modules/core/include/opencv2/core/quaternion.hpp

```diff
@@ -0,0 +1,1949 @@
+         * @note Due to the tangent, the quaternion cannot convert to rotation vector when the rotation angle equals \f$pi + 2kpi\f$ radians or \f$180 + 360k\f$ degree, \f$k = ... -2, -1, 1, 2 ...\f$.
+         */
+        cv::Vec<_Tp, 3> toRodrigues(bool assumeUnit=false)  const;
+
+        /**
+        * @brief get the angle of quaternion, it returns the rotate angle.
+        * @param assumeUnit if true, this quaternon should be a unit quaternion and
+        * this function will save some computations.
+        * \f[\psi = 2 *arccos(\frac{w}{||q||})\f]
+        *
+        * For example
+        * ```
+        * Quatd q(1,2,3,4);
+        * q.getAngle();
+        *
+        * q.normalize().getAngle(true);//same as q.getAngle()
+        * ```
+        */
+        _Tp getAngle(bool assumeUnit=false) const;
+
+        /**
+        * @brief get the axis of quaternion, it returns a vector of length 3
+        * @param assumeUnit if true, this quaternon should be a unit quaternion and
+        * this function will save some computations.
+        * the unit axis \f$\boldsymbol{n}\f$ is defined by
+        * \f[\begin{array}
+        *      \boldsymbol{v} &= \boldsymbol{u} ||\boldsymbol{v}||\\
+        *             &= \boldsymbol{u}||q||sin(\theta)
+        *      \end{array}\f]
+        *
+        * For example
+        * ```
+        * Quatd q(1,2,3,4);
+        * q.getAxis();
+        *
+        * q.normalize().getAxis(true);//same as q.getAxis()
+        * ```
+        */
+        cv::Vec<_Tp, 3> getAxis(bool assumeUnit=false) const;
+
+        /**
+         * @brief return the dot between quaternion \f$q\f$ and this quaternion, i.e.,
+         * \f[p \cdot q = this.w \cdot q.w + this.x \cdot x + this.y \cdot y + this.z \cdot z\f]
+         * @param q the other quaternion
+         *
+         * For example
+         * ```
+         * Quatd q(1,2,3,4);
+         * Quatd p(5,6,7,8);
+         * p.dot(q);
+         * ```
+         */
+        _Tp dot(Quat<_Tp> q) const;
+
+        /**
+         * @brief To calculate the interpolation from \f$q_0\f$ to \f$q_1\f$ by Linear Interpolation(Nlerp)
+         * For two quaternions, this interpolation curve can be displayed as:
+         * \f[Lerp(q_0, q_1, t) = (1 - t)q_0 + tq_1 \f]
+         * Obviously, the lerp will interpolate along a straight line if we think of \f$q_0\f$ and \f$q_1\f$ as a vector
+         * in a two-demensional space. When \f$t = 0\f$, it returns \f$q_0\f$ and when \f$t= 1\f$, it returns \f$q_1\f$.
+         * \f$t\f$ should be ranged in \f$[0, 1]\f$ normally
+         * @param q0 a quaternion used in linear interpolation
+         * @param q1 a quaternion used in linear interpolation
+         * @param t percent of vector \f$\overrightarrow{q_0q_1}\f$ over a range [0, 1]
+         * @note it returns a non-unit quaternion.
+         */
+        static Quat<_Tp> lerp(const Quat<_Tp> &q0, const Quat &q1, const _Tp t);
```

Comment:
What is the purpose to make this function static, but not free function?

Reply:
If it's free function, user can't find it in documentation